common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -803,6 +803,8 @@ def _get_answer_notification(self, render_notifications):
                 else:
                     answer_notification_message = _('Correct')
             elif answer_notification_type == 'partially-correct':
+                # representing values as a whole numbers
+                progress = Progress(round(progress.frac()[0], 1), progress.frac()[1])
```

Comment:
The problem is that we're allowing arbitrary points for the denominator. For example, you can make a problem whose total score is 3.14159 points. In that case, we don't really want to round the numerator or the denominator.

If you look [here](https://github.com/edx/edx-platform/blob/c4c5a8e8a9cc2ae4bdf908c2c26150dc699c3b58/common/lib/capa/capa/responsetypes.py#L722-L722), you'll see that `edc_current_grade` and `edc_max_grade` are always going to be integers. At that point, there are two possibilities: 

1. `self.get_max_score()` returns an integer. In that case, return_grade can be a [`Fraction`](https://docs.python.org/2/library/fractions.html) type. You can keep passing that around, and convert it into a float at the last possible minute. This route will require a good deal of refactoring.

2. `self.get_max_score()` returns a float. At this point, the logic should continue as is.

I think there's too much float multiplication and rounding in the logic to fix this bug by adding more rounding. I personally think that we should file a story to refactor [Progress](https://github.com/edx/edx-platform/blob/dbb66f8c099f5e496a74449ed6d6bdc7f49f3d5b/common/lib/xmodule/xmodule/progress.py#L11-L11) to inherit from python's built in `Fraction` type. It may be worthwhile for us to pick this story up, but we would let product prioritize it.

Happy to talk about this more tomorrow!